gateway/mw_url_rewrite_test.go

```diff
@@ -1111,6 +1146,14 @@ func TestRewriterTriggers(t *testing.T) {
 			if err != nil {
 				t.Error("compile failed:", err)
 			}
+
+			//added check to ensure that reading the payload to check for the trigger does not break the request
+			if tc.payloadTrigger {
+				body, err := io.ReadAll(tc.req.Body)
+				assert.NotEqual(t, "", string(body))
+				assert.Nil(t, err)
```

Comment:
```suggestion
				assert.NoError(t, err)
```

Nil is for *ptr results, while Error/NoError are for typed error checks